awx/main/utils/external_logging.py

```diff
@@ -0,0 +1,65 @@
+import urllib.parse as urlparse
+
+from django.conf import settings
+
+from awx.main.utils.reload import supervisor_service_command
+
+
+def reconfigure_rsyslog():
+    tmpl = ''
+    if settings.LOG_AGGREGATOR_ENABLED:
+        host = getattr(settings, 'LOG_AGGREGATOR_HOST', '')
+        port = getattr(settings, 'LOG_AGGREGATOR_PORT', '')
+        protocol = getattr(settings, 'LOG_AGGREGATOR_PROTOCOL', '')
+
+        if protocol.startswith('http'):
+            scheme = 'https'
+            # urlparse requires '//' to be provided if scheme is not specified
+            original_parsed = urlparse.urlsplit(host)
+            if (not original_parsed.scheme and not host.startswith('//')) or original_parsed.hostname is None:
+                host = '%s://%s' % (scheme, host) if scheme else '//%s' % host
+            parsed = urlparse.urlsplit(host)
+
+            host = parsed.hostname
+            try:
+                port = parsed.port or settings.LOG_AGGREGATOR_PORT
+            except ValueError:
+                port = settings.LOG_AGGREGATOR_PORT
+
+        parts = []
+        parts.extend([
+            '$ModLoad imudp',
+            '$UDPServerRun 51414',
+            'template(name="awx" type="string" string="%msg%")',
+        ])
+        if protocol.startswith('http'):
+            # https://github.com/rsyslog/rsyslog-doc/blob/master/source/configuration/modules/omhttp.rst
+            ssl = "on" if parsed.scheme == 'https' else "off"
+            skip_verify = "off" if settings.LOG_AGGREGATOR_VERIFY_CERT else "on"
+            params = [
+                'type="omhttp"',
+                f'server="{host}"',
+                f'serverport="{port}"',
+                f'usehttps="{ssl}"',
+                f'skipverifyhost="{skip_verify}"',
+                'action.resumeRetryCount="-1"',
+                'template="awx"',
+                'errorfile="/var/log/tower/external.err"',
+            ]
+            username = getattr(settings, 'LOG_AGGREGATOR_USERNAME', '')
+            password = getattr(settings, 'LOG_AGGREGATOR_PASSWORD', '')
+            if username:
+                params.append(f'uid="{username}"')
+            if password:
+                params.append(f'pwd="{password}"')
+            params = ' '.join(params)
+            parts.extend(['module(load="omhttp")', f'action({params})'])
+        else:
+            parts.append(
+                f'action(type="omfwd" target="{host}" port="{port}" protocol="{protocol}" action.resumeRetryCount="-1" template="awx")'  # noqa
+            )
```

Comment:
Currently, if the config is invalid (which is the case if logging is disabled because the rsyslog.conf will just be empty), we get:
```
bash-4.4$ supervisorctl status tower-processes:awx-rsyslogd
tower-processes:awx-rsyslogd     FATAL     Exited too quickly (process log may have details)
```

I am trying to find a default config that will allow rsyslog to remain running when logging is disabled.

Reply:
I chatted with @ryanpetrello about this and we are OK with this as long as it doesn't get stuck in a restart loop (it doesn't) and as long as it goes back in to the RUNNING state properly when restarted with a valid config, which is does.